src/plugins/kibana_react/public/saved_objects/saved_object_finder.tsx

```diff
@@ -0,0 +1,520 @@
+    fixedPageSize: PropTypes.number,
+    showFilter: PropTypes.bool,
+  };
+
+  private isComponentMounted: boolean = false;
+
+  private debouncedFetch = _.debounce(async (query: string) => {
+    const metaDataMap = this.getSavedObjectMetaDataMap();
+
+    const perPage = this.props.uiSettings.get('savedObjects:listingLimit');
+    const resp = await this.props.savedObjects.client.find({
+      type: Object.keys(metaDataMap),
+      fields: ['title', 'visState'],
+      search: query ? `${query}*` : undefined,
+      page: 1,
+      perPage,
+      searchFields: ['title^3', 'description'],
+      defaultSearchOperator: 'AND',
+    });
+
+    resp.savedObjects = resp.savedObjects.filter(savedObject => {
+      const metaData = metaDataMap[savedObject.type];
+      if (metaData.showSavedObject) {
+        return metaData.showSavedObject(savedObject);
+      } else {
+        return true;
+      }
+    });
+
+    if (!this.isComponentMounted) {
+      return;
+    }
+
+    // We need this check to handle the case where search results come back in a different
+    // order than they were sent out. Only load results for the most recent search.
+    if (query === this.state.query) {
+      this.setState({
+        isFetchingItems: false,
+        items: resp.savedObjects.map(savedObject => {
+          const {
+            attributes: { title },
+            id,
+            type,
+          } = savedObject;
+          return {
+            title: typeof title === 'string' ? title : '',
+            id,
+            type,
+            savedObject,
+          };
+        }),
+      });
+    }
+  }, 300);
+
+  constructor(props: SavedObjectFinderProps) {
+    super(props);
+
+    this.state = {
+      items: [],
+      isFetchingItems: false,
+      page: 0,
+      perPage: props.initialPageSize || props.fixedPageSize || 10,
+      query: '',
+      filterOpen: false,
+      filteredTypes: [],
+      sortOpen: false,
+    };
+  }
+
+  public componentWillUnmount() {
+    this.isComponentMounted = false;
+    this.debouncedFetch.cancel();
+  }
+
+  public componentDidMount() {
+    this.isComponentMounted = true;
+    this.fetchItems();
+  }
+
+  public render() {
+    return (
+      <React.Fragment>
+        {this.renderSearchBar()}
+        {this.renderListing()}
+      </React.Fragment>
+    );
+  }
+
+  private getSavedObjectMetaDataMap(): Record<string, SavedObjectMetaData<SavedObjectAttributes>> {
+    return this.props.savedObjectMetaData.reduce(
+      (map, metaData) => ({ ...map, [metaData.type]: metaData }),
+      {}
+    );
+  }
+
+  private getPageCount() {
+    return Math.ceil(
+      (this.state.filteredTypes.length === 0
+        ? this.state.items.length
+        : this.state.items.filter(
+            item =>
+              this.state.filteredTypes.length === 0 || this.state.filteredTypes.includes(item.type)
+          ).length) / this.state.perPage
+    );
+  }
+
+  // server-side paging not supported
```

Comment:
Is this comment still accurate?

Reply:
I don't know, I haven't modified or verified this code.